apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -1701,32 +1728,32 @@ send_trailer(StateData) ->
     send_text(StateData, ?STREAM_TRAILER).
 
 
-send_and_maybe_buffer_stanza({J1, J2, El}, State, StateName)->
-    % this is a very last stage, we already terminated accumulator
+-spec send_and_maybe_buffer_stanza(mongoose_acc:t(), packet(), state()) ->
+    {ok | resume, mongoose_acc:t(), state()}.
+send_and_maybe_buffer_stanza(Acc, {J1, J2, El}, State)->
     % to be removed
     {SendResult, BufferedStateData} =
         send_and_maybe_buffer_stanza({J1, J2, mod_amp:strip_amp_el_from_request(El)}, State),
-    % if we have to check packet before sending we should do it much earlier, when it is
-    % still accumulator
     mod_amp:check_packet(El, result_to_amp_event(SendResult)),
```

Comment:
Because mod_amp doesn't yet know what to do with it. It certainly should - now that we have accumulators mod_amp doesn't need to use stanza to carry internal info. But rewriting mod_amp is beyond the scope of this project.